UndertaleModTool/Scripts/Community Scripts/TouchControlsEnabler.csx

```diff
@@ -100,11 +107,48 @@ ImportGMLFile(Path.Combine(dataPath, "gml_Object_obj_mobilecontrols_Step_0.gml")
 var mobileControls = Data.GameObjects.ByName("obj_mobilecontrols");
 mobileControls.Persistent = true;
 
-if (displayName == "deltarune chapter 1 & 2" || displayName == "deltarune chapter 1&2")
+if(otherName == false)
 {
-    Data.Code.ByName("gml_Object_obj_gamecontroller_Create_0").AppendGML("instance_create(0, 0, obj_mobilecontrols);", Data);
+    if (displayName == "deltarune chapter 1 & 2" || displayName == "deltarune chapter 1&2")
+    {
+        Data.Code.ByName("gml_Object_obj_gamecontroller_Create_0").AppendGML("instance_create(0, 0, obj_mobilecontrols);", Data);
+    }
+    else if (displayName == "undertale") 
+    {
+        Data.Code.ByName("gml_Object_obj_time_Create_0").AppendGML("instance_create(0, 0, obj_mobilecontrols);", Data);
+    }
 }
-else if (displayName == "undertale") 
+else
 {
-    Data.Code.ByName("gml_Object_obj_time_Create_0").AppendGML("instance_create(0, 0, obj_mobilecontrols);", Data);
+    var obj_time = Data.GameObjects.ByName("obj_time");
+    if(obj_time != null)
+    {
+        Data.Code.ByName("gml_Object_obj_time_Create_0").AppendGML("instance_create(0, 0, obj_mobilecontrols);", Data);
+        return;
+    }
+    var obj_gamecontroller = Data.GameObjects.ByName("obj_gamecontroller");
+    if(obj_gamecontroller != null)
+    {
+        Data.Code.ByName("gml_Object_obj_gamecontroller_Create_0").AppendGML("instance_create(0, 0, obj_mobilecontrols);", Data);
+        return;
+    }
+    var firstRoom = Data.Rooms[0];
+    var shouldAdd = true;
+
+    foreach(var room_obj in firstRoom.GameObjects) {
+        if(room_obj.ObjectDefinition == mobileControls) {
+            shouldAdd = false;
+        }
+    }
```

Comment:
You can use a linq statement here. something along the lines of `shouldAdd = firstRoom.GameObjects.Any(obj => obj,ObjectDefinition == mobileControls)`